test/e2e/specs/site-editor/template-revert.spec.js

```diff
@@ -0,0 +1,272 @@
+
+		const contentAfter = await templateRevertUtils.getCurrentSiteEditorContent();
+		expect( contentBefore ).toEqual( contentAfter );
+	} );
+
+	test( 'should show the original content after revert, clicking undo then redo in the header toolbar', async ( {
+		page,
+		templateRevertUtils,
+	} ) => {
+		const contentBefore = await templateRevertUtils.getCurrentSiteEditorContent();
+
+		await templateRevertUtils.addDummyText();
+		await templateRevertUtils.save();
+		await templateRevertUtils.revertTemplate();
+		await templateRevertUtils.save();
+		const headerToolbar = page.locator( 'role=region[name="Header"]' );
+		await headerToolbar.locator( '[aria-label="Undo"]' ).click();
+
+		const contentAfterUndo = await templateRevertUtils.getCurrentSiteEditorContent();
+		expect( contentBefore ).not.toEqual( contentAfterUndo );
+
+		const redoButton = headerToolbar.locator( '[aria-label="Redo"]' );
+
+		await redoButton.click();
+
+		const contentAfterRedo = await templateRevertUtils.getCurrentSiteEditorContent();
+		expect( contentBefore ).toEqual( contentAfterRedo );
+	} );
+
+	test( 'should show the original content after revert, clicking undo in the notice then undo in the header toolbar', async ( {
+		page,
+		templateRevertUtils,
+	} ) => {
+		const contentBefore = await templateRevertUtils.getCurrentSiteEditorContent();
+
+		await templateRevertUtils.addDummyText();
+		await templateRevertUtils.save();
+		await templateRevertUtils.revertTemplate();
+		await templateRevertUtils.save();
+
+		// Click undo in the snackbar. This reverts revert template action.
+		const snackBar = page.locator(
+			'role=button[name="Dismiss this notice"]'
+		);
+		await snackBar.locator( 'role=button >> text="Undo"' ).click();
+
+		//Check we have dummy content.
+		const contentAfterFirstUndo = await templateRevertUtils.getCurrentSiteEditorContent();
+		expect( contentBefore ).not.toEqual( contentAfterFirstUndo );
+
+		// Click undo again, this time in the header. Reverts initial dummy content.
+		const headerToolbar = page.locator( 'role=region[name="Header"]' );
+		await headerToolbar.locator( '[aria-label="Undo"]' ).click();
+
+		// Check dummy content is gone.
+		const contentAfterSecondUndo = await templateRevertUtils.getCurrentSiteEditorContent();
+		expect( contentBefore ).toEqual( contentAfterSecondUndo );
+	} );
+
+	test( 'should show the edited content after revert, clicking undo in the header toolbar, save and reload', async ( {
+		admin,
+		page,
+		templateRevertUtils,
+	} ) => {
+		await templateRevertUtils.addDummyText();
+		await templateRevertUtils.save();
+		const contentBefore = await templateRevertUtils.getCurrentSiteEditorContent();
+
+		await templateRevertUtils.revertTemplate();
+		await templateRevertUtils.save();
+
+		const headerToolbar = page.locator( 'role=region[name="Header"]' );
+		await headerToolbar.locator( '[aria-label="Undo"]' ).click();
+
+		await templateRevertUtils.save();
+
+		await admin.visitSiteEditor();
```

Comment:
Is there any reason why we can't just do a `page.reload()`?

Reply:
The test fails when I replace that line with `await page.reload()`, with a diff between `contentBefore` and `contentAfter`. Funny thing is, if I move the assertion to before the page reload, the test passes. Not at all sure what's happening there, because when I run it in debug mode, it always passes 😅

Reply:
Ahh, okay, I guess we have to wait for the canvas to be loaded. Might be simpler to just do `visitSiteEditor()` then.